schema-registry/common/src/main/java/com/hortonworks/registries/schemaregistry/avro/AvroFieldsGenerator.java

```diff
@@ -67,36 +71,47 @@ private void parseField(Schema.Field field, List<SchemaFieldInfo> schemaFieldInf
         schemaFieldInfos.add(new SchemaFieldInfo(namespace, name, type.name()));
 
         // todo check whether fields should be mapped to the root schema.
-        parseSchema(schema, schemaFieldInfos);
+        parseSchema(schema, schemaFieldInfos, visitedRecords);
     }
 
-    private void parseSchema(Schema schema, List<SchemaFieldInfo> schemaFieldInfos) {
+    private void parseSchema(Schema schema, List<SchemaFieldInfo> schemaFieldInfos, Set<String> visitedRecords) {
         Schema.Type type = schema.getType();
         LOG.debug("Visiting type: [{}]", type);
 
         switch (type) {
             case RECORD:
-                // store fields of a record.
-                List<Schema.Field> fields = schema.getFields();
-                for (Schema.Field recordField : fields) {
-                    parseField(recordField, schemaFieldInfos);
+
+                String completeName = schema.getFullName();
+
+                if (visitedRecords.contains(completeName)) {
```

Comment:
Since the if branch is a no-op, can be rewritten as:
```java
if (!visitedRecords.contains(completeName)) {
  visitedRecords.add(completeName);
  // store fields of a record.
  ..
  ..
}
```

Reply:
Oh duh, how'd I miss that. Thanks, will update!